app/src/main/java/ai/elimu/appstore/ui/applications/ApplicationListActivity.java

```diff
@@ -65,13 +65,21 @@ protected void onStart() {
         RoomDb.databaseWriteExecutor.execute(() -> {
             List<Application> applications = applicationDao.loadAll();
             Log.d(getClass().getName(), "applications.size(): " + applications.size());
-            applicationListAdapter.setApplications(applications);
-
+            if(applications != null) {
+                applicationListAdapter.setApplications(applications);
+            }
             List<ApplicationVersion> applicationVersions = applicationVersionDao.loadAll();
             Log.d(getClass().getName(), "applicationVersions.size(): " + applicationVersions.size());
             applicationListAdapter.setApplicationVersions(applicationVersions);
 
-            applicationListAdapter.notifyDataSetChanged();
+
+            recyclerView.post(new Runnable() {
```

Comment:
To avoid the dependence on `recyclerView`, use `runOnUiThread` instead?:

```suggestion
            runOnUiThread(new Runnable() {
```